component-runtime-beam/src/main/java/org/talend/sdk/component/runtime/beam/spi/record/AvroRecord.java

```diff
@@ -104,6 +105,10 @@ private Object directMapping(final Object value) {
         if (value instanceof Date) {
             return Date.class.cast(value).getTime();
         }
+        // those convert here is necessary?
+        if (value instanceof BigDecimal) {
+            return BigDecimal.class.cast(value).toString();
```

Comment:
Not clear to me if this feature should support exponential notation,
If yes, add a unit test with something like "1.23E-15"
if no, use BigDecimal.toPlainString instead toString method

Reply:
only new AvroRecord(RecordImpl instance) can go into this code in AvroRecord, can this code can make recordimpl which can affect here:
```
RecordImpl.BuilderImpl.with(Entry with tck Decimal type, BigDecimal) 
```
And will check it